compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -27,14 +27,38 @@ use rustc_span::Span;
 /// employing the ExprUseVisitor.
 pub trait Delegate<'tcx> {
     // The value found at `place` is either copied or moved, depending
-    // on mode.
-    fn consume(&mut self, place_with_id: &PlaceWithHirId<'tcx>, mode: ConsumeMode);
+    // on mode. Where `diag_expr_id` is the id used for diagnostics for `place`.
```

Comment:
This doesn't seem grammatically correct as a sentence. Perhaps:

The value found at `place` is either copied or moved, depending on `mode`. 

The parameter `diag_expr_id` indicates the HIR id that ought to be used for diagnostics. Around pattern matching such as `let pat = expr`, the diagnostic id will be the id of the expression `expr` but the place itself will have the id of the binding in the pattern `pat`.